src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseStudyServiceImpl.java

```diff
@@ -1021,23 +1021,25 @@ public T findByIdInAnyTenant(final Long studyId) {
         return studyRepository.findByIdInAnyTenant(studyId);
     }
 
-    @PostAuthorize("@ArachnePermissionEvaluator.addPermissions(principal, returnObject )")
     @Override
-    public T findWorkspaceForUser(Long userId) throws NotExistException {
+    public T findWorkspaceForUser(IUser user, Long userId) throws NotExistException {
 
```

Comment:
But why user.getId() is not enough?

Reply:
because we should check permissions and it will be done in getStudy method

Reply:
Did you mean getStudy call inside this method?

Reply:
ofc

Reply:
also we can get current logged user from context... maybe it will be lil clearer